### PR TITLE
hotel management app and its APIs

### PR DESCRIPTION
**Pull request by harmeshverma01**

### CODE REVIEW

File: room/views.py [hunk 1]
```diff
@@ -1,28 +1,58 @@
 from django.shortcuts import render
```
Comment (aayush-kumar-codes) [inline room/views.py:1]: remove this import

File: room/views.py [hunk 2]
```diff
@@ -1,28 +1,58 @@
...
+from room.models import Room
 from rest_framework.response import Response
 from rest_framework import status
+
 # Create your views here.
 
 
 class Roomview(APIView):
     serializer_class = RoomSerializer
+
```
Comment (aayush-kumar-codes) [inline room/views.py:14]: remove extra spaces

File: room/views.py [hunk 3]
```diff
@@ -1,28 +1,58 @@
...
+                serializer.save()
+            return Response(serializer.data)
+        except Room.DoesNotExist:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+        
+        
+    def delete(self, request, id=None):
+       user = Room.objects.get(id=id)
+       user.delete()
+       return Response(({"message": "Room is deleted"}),status=status.HTTP_204_NO_CONTENT)
```
Comment (aayush-kumar-codes) [inline room/views.py:57]: formate the code.

File: user/urls.py [hunk 4]
```diff
@@ -1,8 +1,8 @@
...
-from .views import Userview, Loginview
+from .views import UserDetailsView, Userview, Loginview
 
 urlpatterns = [
-    path('users/', Userview.as_view()),
-    path('login/', Loginview.as_view())
-    
+    path('user/', Userview.as_view()),
+    path('login/', Loginview.as_view()),
+    path('userdetails/', UserDetailsView.as_view())
```
Comment (aayush-kumar-codes) [inline user/urls.py:7]: No need to create endpoint with separate name. Just send the id in the url.

File: user/views.py [hunk 5]
```diff
@@ -1,20 +1,23 @@
...
+from rest_framework import authentication, permissions
 from rest_framework.response import Response
 from rest_framework import status
 from .serializers import UserSerializer
 from .models import User
 from django.contrib.auth import authenticate
 from rest_framework.authtoken.models import Token
+from .utils import admin_required
+
  # Create your views here.
```
Comment (aayush-kumar-codes) [inline user/views.py:11]: remove it

File: user/views.py [hunk 6]
```diff
@@ -55,4 +58,29 @@ def post(self, request):
...
+            if serializer.is_valid():
+                serializer.save()
+            return Response(serializer.data)
+        except User.DoesNotExist:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+        
+    def delete(self, request, id=None):
+       user = User.objects.get(id=id)
+       user.delete()
+       return Response(({"message": "User is deleted"}),status=status.HTTP_204_NO_CONTENT)
```
Comment (aayush-kumar-codes) [inline user/views.py:85]: format correctly

File: user/views.py [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+        user = User.objects.create_user(
+            name = request.data.get('name'),
+            email = request.data.get('email'),
+            password = request.data.get('password'),
+        )
+        serializer =  UserSerializer(user)
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
+    
+    
+
```
Comment (aayush-kumar-codes) [inline user/views.py:35]: only one line space

File: room/views.py (hunk 1 above)
Comment (harmeshverma01) [reply room/views.py:1]: fixed

File: room/views.py (hunk 2 above)
Comment (harmeshverma01) [reply room/views.py:14]: fixed

File: user/views.py [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+    
+    
+    
+    def patch(self, request, id=None):
+        try:
+            user = User.objects.get(id=id)
+            serializer = self.serializer_class(user, data=request.data, partial=True)
+            if serializer.is_valid():
+                serializer.save()
+            return Response(serializer.data)
```
Comment (aayush-kumar-codes) [inline user/views.py:42]: what will happen if serializer not valid?

File: user/urls.py (hunk 4 above)
Comment (harmeshverma01) [reply user/urls.py:7]: fixed

File: user/views.py (hunk 5 above)
Comment (harmeshverma01) [reply user/views.py:11]: fixed

File: user/views.py (hunk 7 above)
Comment (harmeshverma01) [reply user/views.py:35]: fixed

File: user/views.py (hunk 6 above)
Comment (harmeshverma01) [reply user/views.py:85]: fixed

File: user/views.py (hunk 8 above)
Comment (harmeshverma01) [reply user/views.py:42]: fixed

**Pull request merged**